integration-tests-support/certificate/src/main/java/org/apache/camel/quarkus/test/support/certificate/TestCertificateGenerationExtension.java

```diff
@@ -0,0 +1,143 @@
+
+        extensionContext.getStore(ExtensionContext.Namespace.GLOBAL)
+                .getOrComputeIfAbsent(TestCertificateGenerationExtension.class, c -> this);
+        var maybe = AnnotationUtils.findAnnotation(extensionContext.getRequiredTestClass(), TestCertificates.class);
+        if (maybe.isEmpty()) {
+            return;
+        }
+        var annotation = maybe.get();
+
+        //cn and alternativeSubjectName might be different (to reflect docker host)
+        Optional<String> cn = resolveDockerHost();
+        Optional<String> altSubName = cn.stream().map(h -> "DNS:%s,IP:%s".formatted(h, h)).findAny();
+
+        for (Certificate certificate : annotation.certificates()) {
+            String baseDir = annotation.baseDir();
+            File file = new File(baseDir);
+            file.mkdirs();
+            CertificateGenerator generator = new CertificateGenerator(file.toPath(), annotation.replaceIfExists());
+
+            CertificateRequest request = new CertificateRequest()
+                    .withName(certificate.name())
+                    .withClientCertificate(certificate.client())
+                    .withFormats(Arrays.asList(certificate.formats()))
+                    .withCN(cn.orElse(certificate.cn()))
+                    .withPassword(certificate.password().isEmpty() ? null : certificate.password())
+                    .withDuration(Duration.ofDays(certificate.duration()));
+
+            if (cn.isPresent() && cn.get().equals(certificate.cn())) {
+                LOGGER.debugf("Used CN '%s' instead of '%s' because of docker host.", cn.get(), certificate.cn());
+                LOGGER.debugf("Added SubjectAlternativeName '%s'.", altSubName.get());
+            }
+
+            if (altSubName.isPresent()) {
+                request.withSubjectAlternativeName(altSubName.get());
+            }
+
+            for (String san : certificate.subjectAlternativeNames()) {
+                request.withSubjectAlternativeName(san);
+            }
+
+            for (Alias alias : certificate.aliases()) {
+                AliasRequest nested = new AliasRequest()
+                        .withCN(alias.cn())
+                        .withPassword(alias.password())
+                        .withClientCertificate(alias.client());
+                request.withAlias(alias.name(), nested);
+                for (String s : alias.subjectAlternativeNames()) {
+                    nested.withSubjectAlternativeName(s);
+                }
+            }
+
+            certificateFiles.addAll(generator.generate(request));
+        }
+    }
+
+    private Optional<String> resolveDockerHost() {
+        String dockerHost = DockerClientFactory.instance().dockerHostIpAddress();
+        if (!dockerHost.equals("localhost") && !dockerHost.equals("127.0.0.1")) {
+            String imageName = ConfigProvider.getConfig().getValue("eclipse-temurin.container.image", String.class);
```

Comment:
The stuff that was using `eclipse-temurin.container.image` I assume is not needed anymore given that the cert generator will take care of what it was being used for?

So maybe any references to this method can just be replaced with `DockerClientFactory.instance().dockerHostIpAddress()`?

Reply:
I started container containe to access method `container.getHost()`. I was not sure if `DockerClientFactory.instance().dockerHostIpAddress()` returns the same value. Probably yes, in that case I'll remove the container.